modules/channel-web/src/views/lite/main.tsx

```diff
@@ -62,6 +63,22 @@ class Web extends React.Component<MainProps> {
   }
 
   componentDidUpdate() {
+    const app = document.querySelector('#app')
+    const div = app?.firstChild?.firstChild
+    if (div) {
+      div.childNodes.forEach(child => {
+        const href = child['attributes']?.href
+        if (href) {
+          const value = href?.value
+          if (value && typeof value === 'string') {
+            const foundFontCss = value.match(/font-roboto.css/)?.length > 0
+            setTimeout(() => {
+              this.hasFontCss = foundFontCss
+            }, 500)
+          }
+        }
+      })
+    }
```

Comment:
This should definitely be put in a function explaining its purpose ... 
With a lot of comments. 
Too many indentations.

Reply:
Well, this points to an HTML element on the DOM and iterates on its children DOM elements. If any of those elements has an attribute `href`, we pull its value, check if it's a string and match it against a RegExp that verifies if it's font-roboto.css. We then assign the boolean value to state.

It helps not displaying an awkward Botpress design for the embedded chat icon when the site is not completely loaded yet. It's only perceptible for users with bad internet connections. This, coupled with my change below where the stylesheet is served first; helps in this respect.

Unfortunately I haven't found a way to *completely* resolve the issue, but it's *much* better than it was.